New Pattern: Code of Conduct (#661)
gruetter/innerSourceTest: patterns/1-initial/code-of-conduct.md

> ## Title

Reviewer: collection of general feedback here:

- Is there a general description of the goals of a CoC that we can link to?
- What happens when the CoC is broken? Is that something that could be added to the solution or implementation section? Here is an [a conference talk](https://www.youtube.com/watch?v=RBZrYMEJFXA&list=PLq-odUc2x7i9Zrs4oyM1I3V5xlYICEL2g&index=8) related to this that I heard once, and also the [Mozilla Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/) contain something about the reporting process in case of a CoC violation. Again this point might be different in InnerSource than in open source, and therefore even more valuable to describe as part of this pattern.

Author: Answering the `Is there a general description of the goals of a CoC that we can link to?` question, this [reference](https://opensource.guide/code-of-conduct/) is a good one from the Open Source space, but I don't see how to link here. Maybe adding this reference in the Related patterns...

Regarding to broken the CoC, the implementation section is extended including the reporting process and the consequences when the CoC is not followed. It is a general description, just because, each CoC can differ in the final implementation of these processes, however, all of them must be part of the CoC.

Reviewer: I think it doesn't hurt to add it under References at the end of this pattern.

Actually i renamed "Related Patterns" to "References" and added the link in. Please take a look if you like the approach.

Code of Conduct

## Patlet

> A healthy and inclusive InnerSource community requires strong levels of communications and interactions.

Reviewer: If you were to rewrite the Patlet as 2 sentences (1 problem + 1 solution), how might that look like?

Establishing a Code of Conduct provides guidelines for establishing rules and expectations regarding behavior and interactions within the community.

## Problem

InnerSource initiatives often involve collaboration among individuals and teams from diverse backgrounds and perspectives.
The absence of clear rules of communication and interaction leaves room for ambiguity and potential conflicts within the InnerSource community.
Without established guidelines, members may engage in behavior that is harmful, discriminatory, or counterproductive, leading to a breakdown in collaboration and trust.

> ## Context

Reviewer: The Context section should describe the following:

> Where does the problem exist? What are the pre-conditions? Unchangeable before the solution goes into place. The content here is often tied to the applicability of the pattern for other readers: "Do I have this same particular situation?"

The current text rather describes the state that should be achieved after the pattern has been implemented, I think?

Author: Does this new content fit better the Context section?

Reviewer: There is no right or wrong here, really. However the current Contest ist not what I had in mind. Will try to provide better examples.

Often times the Context section will describe the situation at a typical organization at which this issue exists.

e.g. for this pattern I could imagine thinks like

- the organization has gone through a rapid growth phase, where maintaining a strong collaborative culture has suffered
- the org has acquired multiple smaller organizations over a short period of time. each of these org brings their own slightly different culture.
- the organization is highly distributed over multiple countries/nationalities

Author: You are right, adding some scenarios or use cases can help to clarify the context of this pattern. Added your suggestions with some minor changes.

Different individuals and teams within the organization collaborate in a healthy and inclusive community without conflicts.
This welcoming, respectful and inclusive environment fosters more collaboration and creates trusted relationships across the organization where all community members feel safe and valued.

> ## Forces

Reviewer: The Forces section should describe the following:

> What makes the problem difficult? What are the trade-offs? These are constraints that can be changed at a cost. The solution might change one or more of these forces in order to solve the problem, while also in-turn changing the context.

Example of forces that might apply to this pattern could be:

- unclarity by contributors about ....

Author: A listed different actions to clarify better the Forces section. Does it fit better?

Reviewer: As Forces one could also list things like:

- members of the InnerSource community at the organization have not met face to face
- members of the InnerSource community at the organization have not established strong inter-personal relationships
- previous attempts to establish ground rules for collaboration have been treated as mere lip service, and have had little to no effect

Author: I think those sentences can fit better in the Context, as use cases or scenarios, and leave the `Forces` section as is now. What do you think?

Reviewer: For me these are Forces, not Context, as they can be "changed at a cost"

But as I already mentioned before, these two sections are typically the hardest when writing a pattern.
We have not found a good solution to make this easier yet.

So yeah, feel free to leave Forces as is.

Author: Cool! Let's go with this ... let's see go evolve!

A well-crafted Code of Conduct helps prevent and address issues such as harassment, discrimination, and other harmful behavior, ensuring that everyone feels safe and valued within the project.
By providing the framework for positive and inclusive participation, the Code of Conduct encourages diverse perspectives, enhances collaboration, and ultimately leads to the development of higher-quality open-source software.

## Solution

Develop a Code of Conduct that outlines expected behavior, including guidelines for communication, collaboration, and conflict resolution.

> The Code of Conduct will articulate the shared values and principles of the InnerSource community, fostering a sense of belonging and common purpose, such as:

Reviewer: Are the "shared values and principles of the InnerSource community" defined in the CoC somehow tied to the values/principles that the org may have defined in other places already?

If yes, then a note about this could be added to the Implementation section.

Author: I added some references about the integration with the organization rules and employee handbooks. It is the normal way to define these values and principles along the InnerSource project.

Hope that helps to clarify the solution.

- **Diversity**: InnerSource communities may consist of individuals with varying cultural backgrounds, beliefs, and communication styles.
- **Trust**: Building trust among community members is essential for effective collaboration and knowledge sharing.
- **Inclusivity**: A Code of Conduct and Social Contract can promote inclusivity by setting expectations for respectful behavior and interactions.
- **Accountability**: Clear guidelines help hold community members accountable for their actions and contributions.

> ## Implementation

Reviewer: ```suggestion
### Implementation
```

making this a subsection of "Solution", so that we roughly keep the same top level sections in all of our patterns

> The adoption of well-known covenants, such as the [Contributor Covenant](https://www.contributor-covenant.org/), or the adaptation of some internal covenant for members of the organization are good starting points for implementation of this pattern.

Reviewer: Just noting that I, as a non-native English speaker, don't know the word covenant.

Is there a word that means the same but is a bit simpler?
e.g. does "convention" mean the same?

> However, adopt the following life cycle can improve the implementation and adoption of the Code of Conduct in a InnerSource community:

Reviewer: ```suggestion
However, adopting the following life cycle can improve the implementation and adoption of the Code of Conduct in an InnerSource community:
```

1. Collaboratively draft the Code of Conduct, involving input from community members representing diverse perspectives.
2. Seek feedback and consensus from the community to ensure buy-in and ownership of the guidelines.

> 3. Publish the finalized documents in a prominent location accessible to all community members, such as the InnerSource platform or collaboration tools.

Reviewer: Can you elaborate on what you mean by these two things:

- InnerSource platform
- collaboration tool

4. Regularly review and update the Code of Conduct as needed to reflect evolving community norms and values.

> A good practice for the third point is to share the Code of Conduct in each InnerSource community repository as a file named `CODE_OF_CONDUCT.md`.

Reviewer: Is this possibly an area where a CoC in InnerSource could differ from a CoC in open source?

What I mean is, do you expect that different InnerSource projects within an org would end up having different CoCs? If not, could one just link to a central document, to prevent duplication?

Author: I usually see two approaches:
- A shared template repository with the original content to use when a new project starts
- A shared resource linked from each CoC file

To make it easy, just explained the second option. What do you think?

It is important to understand that simply adopting a Code of Conduct will not prevent conflict or toxicity in the InnerSource project.
The [Core Team](https://patterns.innersourcecommons.org/p/core-team) and [Dedicated Community Leader](https://patterns.innersourcecommons.org/p/dedicated-community-leader) are responsible for the safe, fair, and transparent enforcement of the community's code of conduct.
That responsibility will imply provide a reporting process, a gathering information process and the consequences of any unacceptable behavior.
These references must be part of the Code of Conduct to encourage the behavior expected in the InnerSource project.  

## Resulting Context

With a well-defined Code of Conduct in place, the InnerSource community can cultivate a culture of respect, trust, and collaboration.
Community members feel empowered to contribute openly and engage in meaningful dialogue, leading to enhanced innovation and problem-solving.

## Known Instances

- Red Hat - Many of the internal communities
- National Australia Bank

## Authors

- Roman Martin Gil

## Acknowledgments

- Matt Cobby - Adding the National Australia Bank reference

> ## Related Patterns

Reviewer: ```suggestion
## References
```

- [InnerSource Commons Pattern - Standard Base Documentation](../2-structured/base-documentation.md)
- [Open Practice Library - Social Contract](https://openpracticelibrary.com/practice/social-contract/)

> - [Contributor Covenant](https://www.contributor-covenant.org/)

Reviewer: ```suggestion
- [Contributor Covenant](https://www.contributor-covenant.org/)
- [GitHub Open Source Guide: Code of Conduct](https://opensource.guide/code-of-conduct/)
```
